brut.j.util/src/main/java/brut/util/OSDetection.java

```diff
@@ -39,4 +41,26 @@ public static boolean is64Bit() {
     public static String returnOS() {
         return OS;
     }
+    public static boolean isAndroid() {
+        return new File(android_linker).exists();
+    }
+    public static String android_arch() {     
+        if(new File(android_linker).exists()){
+            try{
+                InputStream inputStream = new FileInputStream(android_linker);
+                byte[] bytes = new byte[20];
+                inputStream.read(bytes);
+                if(bytes[18] == (byte)62){
```

Comment:
Is there any documentation anywhere where you got these magic headers?

Reply:
linker64 is the elf linker for android platform. Just like ld-linux.so.2 for glibc based linux platform.
And linker64 itself is an elf format executable. Just lookup elf.h for its header define.
typedef struct
{
  unsigned char	e_ident[EI_NIDENT];	/* Magic number and other info */
  Elf64_Half	e_type;			/* Object file type */
  Elf64_Half	e_machine;		/* Architecture */
  Elf64_Word	e_version;		/* Object file version */
  Elf64_Addr	e_entry;		/* Entry point virtual address */
  Elf64_Off	e_phoff;		/* Program header table file offset */
  Elf64_Off	e_shoff;		/* Section header table file offset */
  Elf64_Word	e_flags;		/* Processor-specific flags */
  Elf64_Half	e_ehsize;		/* ELF header size in bytes */
  Elf64_Half	e_phentsize;		/* Program header table entry size */
  Elf64_Half	e_phnum;		/* Program header table entry count */
  Elf64_Half	e_shentsize;		/* Section header table entry size */
  Elf64_Half	e_shnum;		/* Section header table entry count */
  Elf64_Half	e_shstrndx;		/* Section header string table index */
} Elf64_Ehdr;

Reply:
Here we check e_machine to findout whether it is arm64 or x86_64.
#define EM_X86_64	62	/* AMD x86-64 architecture */
#define EM_AARCH64	183	/* ARM AARCH64 */